scripts/generate-ffdhe.py

```diff
@@ -0,0 +1,514 @@
+    8EC9B55A 7F88A46B 4DB5A851 F44182E1 C68A007E 5E0DD902
+    0BFD64B6 45036C7A 4E677D2C 38532A3A 23BA4442 CAF53EA6
+    3BB45432 9B7624C8 917BDD64 B1C0FD4C B38E8C33 4C701C3A
+    CDAD0657 FCCFEC71 9B1F5C3E 4E46041F 388147FB 4CFDB477
+    A52471F7 A9A96910 B855322E DB6340D8 A00EF092 350511E3
+    0ABEC1FF F9E3A26E 7FB29F8C 183023C3 587E38DA 0077D9B4
+    763E4E4B 94B2BBC1 94C6651E 77CAF992 EEAAC023 2A281BF6
+    B3A739C1 22611682 0AE8DB58 47A67CBE F9C9091B 462D538C
+    D72B0374 6AE77F5E 62292C31 1562A846 505DC82D B854338A
+    E49F5235 C95B9117 8CCF2DD5 CACEF403 EC9D1810 C6272B04
+    5B3B71F9 DC6B80D6 3FDD4A8E 9ADB1E69 62A69526 D43161C1
+    A41D570D 7938DAD4 A40E329C CFF46AAA 36AD004C F600C838
+    1E425A31 D951AE64 FDB23FCE C9509D43 687FEB69 EDD1CC5E
+    0B8CC3BD F64B10EF 86B63142 A3AB8829 555B2F74 7C932665
+    CB2C0F1C C01BD702 29388839 D2AF05E4 54504AC7 8B758282
+    2846C0BA 35C35F5C 59160CC0 46FD8251 541FC68C 9C86B022
+    BB709987 6A460E74 51A8A931 09703FEE 1C217E6C 3826E52C
+    51AA691E 0E423CFC 99E9E316 50C1217B 624816CD AD9A95F9
+    D5B80194 88D9C0A0 A1FE3075 A577E231 83F81D4A 3F2FA457
+    1EFC8CE0 BA8A4FE8 B6855DFE 72B0A66E DED2FBAB FBE58A30
+    FAFABE1C 5D71A87E 2F741EF8 C1FE86FE A6BBFDE5 30677F0D
+    97D11D49 F7A8443D 0822E506 A9F4614E 011E2A94 838FF88C
+    D68C8BB7 C5C6424C FFFFFFFF FFFFFFFF
+'''
+
+# The G value is the same for all parameters, so
+# it is only here once. All the groups from RFC 7919
+# use the generator 2.
+
+g = '''02''' # Not just '2' because these hex strings must
+             # only contain bytes.
+
+def trim(s):
+	'''
+		Remove all whitespace from s and
+		return the resulting string.
+	'''
+	out = ''
+	for c in s:
+		if c == ' ' or c == '\n' or c == '\r':
+			continue
+		out += c
+	return out
+
+def dehexify(hexString):
+	'''
+		Convert the "bignum" hexString
+		representation to a bytearray
+		containing the number.
+	'''
+	hexString = trim(hexString)
+	nbytes = int(len(hexString) / 2)
+	buf = []
+	for idx in range(0, nbytes):
+		buf.append(int(hexString[2 * idx : 2 * idx + 2], 16))
+	return bytearray(buf)
+
+def derLength(size):
+	'''
+		Returns the ASN.1 DER length bytes
+		for the given size.
+
+		Short form: For lengths <= 127,
+		the length is just a byte containing
+		the literal number. (Note that for
+		numbers <= 127, the MSB is not
+		set.)
+
+		Long form: For lengths > 127,
+		the encoding is different: The
+		first byte has the MSB set (that
+		is, by itself, 0x80). The rest of
+		the bits in the first byte specify
+		the number of bytes that follow.
+
+		So, to encode the length 128, you
+		would emit
+
+			0x81 0x80
+
+		The high bit of the first byte is set.
+		This means the remaining 7 bits contain
+		how many bytes follow it. If we mask the
+		MSB away, we end up with 0x01. So, only
+		a single byte follows.
+		The value of the following byte is 0x80,
+		which is 128 decimal.
+	'''
+	if size > 127:
+		hexString = hex(int(size), )[2:]
```

Comment:
Transforms a string to an int then to back to a string in it's hex form. The trailing comma might be a leftover from a removed parameter.